File: LittleGame/LittleGame/TileMaps/Tile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LittleGame.TileMaps
{
    class Tile
    {
        //tile types
        public const int AIR = 0;
        public const int WALL = 1;

        private int tileType;
        private bool blocked;

        public Tile(int type)
        {
            setTileType(type);
        }

        public void setTileType(int type)
        {
            tileType = type;
            if (type == AIR)
            {
                blocked = false;
            }
            else if(type == WALL)
            {
                blocked = true;
            }
        }

        public int getTileType()
        {
            return tileType;
        }

        public bool getBlocked()
        {
            return blocked;
        }
    }
}

File: LittleGameSever_c++/Tile.cpp
#include "Tile.h"
#include <iostream>

Tile::Tile() {}

Tile::Tile(int type = 0)
{
	setTileType(type);
}

int Tile::TileType() { return tileType; }
bool Tile::Blocked() { return blocked; }

void Tile::setTileType(int type)
{
	tileType = type;
	if (type == AIR)
	{
		blocked = false;
	}
	else if (type == WALL)
	{
		blocked = true;
	}
}
